### ISSUE TITLE
Images are very distorted

### ISSUE THREAD

**Issue by khawar-islam** [clovaai/synthtiger#36]
Hello @moonbings During synthetic dataset generation, some image are very distorted and i don't have any idea how to fix it. I played with some parameters but it didnt work for me. **Any solution?**

Samples image are not clear
![23](https://user-images.githubusercontent.com/11488932/196833261-561a9970-ab4b-4b03-acf2-7a3e56f74733.jpg)
![33](https://user-images.githubusercontent.com/11488932/196833271-a03260e1-9119-4ba2-9bb5-0e5ac9c78751.jpg)
![83](https://user-images.githubusercontent.com/11488932/196833278-b5b8a29d-11b3-4216-9346-7d52e410a20f.jpg)

**Comment by moonbings** [comment]
Hi,

This engine often generates poorly images. There are 3 problems in this engine.
Here are some ways to solve this problem.

---

### 1. Color difference threshold

Currently, this engine coverts to grayscale images and checks color differences at the border of text and background. If the color difference does not exceed a given threshold, this image is skipped.
If you adjust the threshold, it can generate better images.
Edit `loDiff`, `upDiff` parameter values in `floodFill` function to `0`, `16` or `0`, `32` in this code.

https://github.com/clovaai/synthtiger/blob/c731f3293130b4c7356429fa2f2b5d17eb6a7382/examples/synthtiger/template.py#L274

### 2. Color difference between text and text effect

The first solution has a limitation.
Currently, this engine does not consider the color difference between text and text effect **(ex2 "그릇")**.
You can solve the problem by adding logic to check the color difference between text and text effect.

### 3. Post-processing

Currently, this engine does not consider post-processing **(ex1 "가?", ex3 "긍정적")**.
The text is not visible due to post-processing, even if the color difference between text and background is large.
If you add color difference checking logic after post-processing, it can genertate better images.
Add the following code in this code.

https://github.com/clovaai/synthtiger/blob/c731f3293130b4c7356429fa2f2b5d17eb6a7382/examples/synthtiger/template.py#L118-L119

```python
if not _check_visibility(image, fg_image[..., 3]):
    raise RuntimeError("Text is not visible")
```

---

I introduced some solutions, but these have not been tested.
Conversely, training performance may be lower.

Thanks.

**Comment by aadit2697** [comment]
@moonbings I was facing a similar issue. Solution 1 you provided has helped to some extent.
Also, I tried out solution 3, but to no avail. 1 in 100 images are distorted and this will affect the training of my data for OCR application. 
Can you suggest a workaround?
Edit: This is an example [ https://user-images.githubusercontent.com/39117677/222642021-e3d40153-ce93-4455-9422-881d91c478ea.jpg ]

the word is: દીધું!


@khawar-islam were you able to solve this issue?

**Comment by khawar-islam** [comment]
@aadit2697 No, I am still facing same issue and of course it affects training

**Comment by aadit2697** [comment]
@khawar-islam 
you want to play around with the style and post-process in the config.yaml  These numbers worked for me. 
Hope this helps! Do let us know if this works out for you :)

style:
  prob: 0.25
  args:
    weights: [1, 2, 2]
    args:
      # text border
      - size: [1, 2]
        alpha: [1, 1]
        grayscale: 0
      # text shadow
      - distance: [1, 2]
        angle: [0, 0]
        alpha: [0.3, 0.7]
        grayscale: 0
      # text extrusion
      - length: [1, 2]
        angle: [0, 360]
        alpha: [1, 1]
        grayscale: 0
postprocess:
  args:
    # gaussian noise
    - prob: 0.0
      args:
        scale: [4, 8]
        per_channel: 0
    # gaussian blur
    - prob: 0.0
      args:
        sigma: [0, 2]
    # resample
    - prob: 0.0
      args:
        size: [0.4, 0.4]
    # median blur
    - prob: 0.0
      args:
        k: [1, 1]

**Comment by khawar-islam** [comment]
@aadit2697 thank you! 

What do you think about two parameters `min_length: 6, max_length: 25 `? I already modified it but sometimes images are distorted. To generate a more clean dataset, if we reduce the max_length, would it be good? What is your opinion?

**Comment by aadit2697** [comment]
> @aadit2697 thank you!
> 
> What do you think about two parameters `min_length: 6, max_length: 25 `? I already modified it but sometimes images are distorted. To generate a more clean dataset, if we reduce the max_length, would it be good? What is your opinion?

You could try it out. Personally, for my use case, I used default values for max and min length. Although If your words are smaller than the 25-character length, you could give it a shot

**Comment by khawar-islam** [comment]
@aadit2697 Did you see anywhere the img_size parameter? I would like to fix it, Are you working on docTR?

**Comment by Tawheed-tariq** [comment]
I also have an issue while generating the images.
The image Quality is good, but for my language the special characters are not getting printed (I have all the special characters in my charset present)

![Image](https://github.com/user-attachments/assets/1030e045-ebe9-4462-b7ed-add338d6952f)

just like in the above image all the special characters are missing

Can you help me to solve this, if you have faced this issue

Thankyou.
